Sources/SourceControl/GitRepository.swift

```diff
@@ -387,6 +397,23 @@ public class GitRepository: Repository, WorkingCheckout {
         return localFileSystem.isDirectory(AbsolutePath(firstLine))
     }
 
+    /// Returns true if the file at `path` is ignored by `git`
+    public func isGitIgnored(_ paths: [AbsolutePath]) throws -> [Bool] {
+        return try queue.sync {
+            let stringPaths = paths.map({ $0.asString })
+            let args = [Git.tool, "-C", self.path.asString, "check-ignore"] + stringPaths
```

Comment:
This might cause issues for two reasons:
1. Paths with spaces needs to be escaped (we should test this).
2. The command-line invocation might surpass ARG_MAX.

I recommend implementing this with git's -z and --stdlib options so we can use a file to provide the input list. See https://git-scm.com/docs/git-check-ignore

Reply:
So we should write the paths to a file and then redirect the file to the stdin of the process and then finally remove the file?

Reply:
Yep! We can use the TemporaryFile class from Basic module.

Reply:
Ok, Thank you!

Reply:
It seems like the redirection isn't working using Process with args. Maybe it's escaping the `<`?